server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1909,6 +1933,10 @@ public Builder putMapping(String source) {
 
         public Builder putMapping(MappingMetadata mappingMd) {
             mapping = mappingMd;
+            Map<String, List<String>> fieldsForModels = mappingMd.getFieldsForModels();
```

Comment:
When mapping is added to the IndexMetadata, check if we can extract the inference models for fields